### PR TITLE
feat(native-security) added flag for controlling security options button visibility

### PR DESCRIPTION
**Pull request by Calinteodor**

### CODE REVIEW

File: react/features/base/flags/constants.js [hunk 1]
```diff
@@ -155,6 +155,12 @@ export const RECORDING_ENABLED = 'recording.enabled';
  */
 export const RESOLUTION = 'resolution';
 
+/**
+ * Flag indicating if the security options button should be enabled.
+ * Default: enabled (true)
+ */
+export const ENABLED_SECURITY_OPTIONS = 'security-options.enabled';
```
Comment (saghul) [inline react/features/base/flags/constants.js:162]: alpha sort please. also, I don't see this used anywhere?

File: react/features/security/components/security-dialog/native/SecurityDialog.js [hunk 2]
```diff
@@ -170,13 +170,39 @@ class SecurityDialog extends PureComponent<Props, State> {
                                 ? 'padding' : 'height'
                         }
                     enabled = { true }>
+                    { this._renderNoSecurityOptions() }
```
Comment (saghul) [inline react/features/security/components/security-dialog/native/SecurityDialog.js:173]: Let's not do this. Instead, let's hide the button if both individual flags are set or if the umbrella flag is set.

**Review by saghul — CHANGES_REQUESTED** [review]

File: react/features/security/components/security-dialog/SecurityDialogButton.js [hunk 3]
```diff
@@ -64,9 +70,13 @@ class SecurityDialogButton extends AbstractButton<Props, *> {
...
     const { locked } = state['features/base/conference'];
     const { lobbyEnabled } = state['features/lobby'];
+    const enabledFlag = getFeatureFlag(state, SECURITY_OPTIONS_ENABLED, true);
+    const enabledLobbyModeFlag = getFeatureFlag(state, LOBBY_MODE_ENABLED, true);
+    const enabledMeetingPassFlag = getFeatureFlag(state, MEETING_PASSWORD_ENABLED, true);
 
     return {
-        _locked: locked || lobbyEnabled
+        _locked: locked || lobbyEnabled,
+        visible: enabledFlag || (enabledLobbyModeFlag && enabledMeetingPassFlag)
```
Comment (saghul) [inline react/features/security/components/security-dialog/SecurityDialogButton.js:79]: You probably intended to use the logic operators in revers: `visible: enabledFlag && (enabledLobbyModeFlag || enabledMeetingPassFlag)`

Note also that lobby might be hidden from base/config or not available (check mapStateToProps in SecurityDialog). We should check for that too here.

File: react/features/base/flags/constants.js (hunk 1 above)
Comment (Calinteodor) [reply react/features/base/flags/constants.js]: Sorted.

File: react/features/security/components/security-dialog/native/SecurityDialog.js (hunk 2 above)
Comment (Calinteodor) [reply react/features/security/components/security-dialog/native/SecurityDialog.js]: Removed it.

File: react/features/security/components/security-dialog/SecurityDialogButton.js (hunk 3 above)
Comment (Calinteodor) [reply react/features/security/components/security-dialog/SecurityDialogButton.js]: Updated condition.

File: react/features/security/components/security-dialog/SecurityDialogButton.js [hunk 4]
```diff
@@ -62,11 +69,19 @@ class SecurityDialogButton extends AbstractButton<Props, *> {
...
  */
 function mapStateToProps(state: Object) {
+    const { conference } = state['features/base/conference'];
+    const { hideLobbyButton } = state['features/base/config'];
     const { locked } = state['features/base/conference'];
     const { lobbyEnabled } = state['features/lobby'];
+    const lobbySupported = conference && conference.isLobbySupported();
+    const lobby = lobbySupported && isLocalParticipantModerator(state) && !hideLobbyButton;
+    const enabledFlag = getFeatureFlag(state, SECURITY_OPTIONS_ENABLED, true);
+    const enabledLobbyModeFlag = getFeatureFlag(state, LOBBY_MODE_ENABLED, true) || lobby;
```
Comment (saghul) [inline react/features/security/components/security-dialog/SecurityDialogButton.js:79]:
```suggestion
    const enabledLobbyModeFlag = getFeatureFlag(state, LOBBY_MODE_ENABLED, true) && lobby;
```

**Review by saghul — APPROVED** [review]

**Pull request merged**